core-services/src/main/java/org/zalando/nakadi/partitioning/MetadataRandomPartitioner.java

```diff
@@ -0,0 +1,45 @@
+package org.zalando.nakadi.partitioning;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.cache.EventTypeCache;
+import org.zalando.nakadi.service.timeline.TimelineService;
+
+import java.util.List;
+import java.util.Random;
+
+@Component
+public class MetadataRandomPartitioner {
+    private final Random random;
+    private final EventTypeCache eventTypeCache;
+    private final TimelineService timelineService;
+
+    @Autowired
+    public MetadataRandomPartitioner(final EventTypeCache eventTypeCache,
+                                     final TimelineService timelineService) {
+        this.random = new Random();
+        this.eventTypeCache = eventTypeCache;
+        this.timelineService = timelineService;
+    }
+
+    public MetadataRandomPartitioner(final EventTypeCache eventTypeCache,
+                                     final TimelineService timelineService,
+                                     final Random random) {
+        this.random = random;
+        this.eventTypeCache = eventTypeCache;
+        this.timelineService = timelineService;
+    }
+
+    public int calculatePartition(final String eventTypeName) {
```

Comment:
returning String here will avoid several conversion to String

Reply:
can we still for now return String to avoid conversion?

Reply:
I removed this class completely, because now we can use PartitionResolver for Avro as well